### PR TITLE
[Batch] Upgrade api-version to v2022-10-01, Add Node'

### PR DESCRIPTION
**Pull request by wiboris**
**Related command**

az batch

**Description**

- Upgrade the Batch data plane to 17.0.0 (API version 2022-10-01.16.0).
- Upgrade the Batch mgmt plane to 13.0.0 (API version 2022-10-01).
- Added new custom enum type `NodeCommunicationMode`.
  - This property determines how a pool communicates with the Batch service.
  - Possible values: Default, Classic, Simplified.
- Added properties `current_node_communication_mode` and `target_node_communication_mode` of type `NodeCommunicationMode` to `CloudPool`.
- Added support for `--target-communication` to pool add and update
- Fixed `az batch account set` command encryption-key-identifier operation

**Testing Guide**

- Testing the Batch module is described in the [README.md](https://github.com/Azure/azurecli/blob/dev/src/azurecli/azure/cli/command_modules/batch/tests/latest/README.md)

**History Notes**
[Batch] `az batch pool create/set`: Add new parameter `--target-communication` to support setting the desired node communication mode for the pool

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Batch

File: src/azure-cli/azure/cli/command_modules/batch/_params.py [hunk 1]
```diff
@@ -238,6 +238,10 @@ def load_arguments(self, _):
                 help='A space separated list of DiskEncryptionTargets. current possible values include OsDisk and TemporaryDisk.', type=disk_encryption_configuration_format)
         c.extra('image', completer=load_supported_images, arg_group="Pool: Virtual Machine Configuration",
                 help="OS image reference. This can be either 'publisher:offer:sku[:version]' format, or a fully qualified ARM image id of the form '/subscriptions/{subscriptionId}/resourceGroups/{resourceGroup}/providers/Microsoft.Compute/images/{imageName}'. If 'publisher:offer:sku[:version]' format, version is optional and if omitted latest will be used. Valid values can be retrieved via 'az batch pool supported-images list'. For example: 'MicrosoftWindowsServer:WindowsServer:2012-R2-Datacenter:latest'")
+        c.argument('target_node_communication_mode', options_list=['--target-communication'])
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/batch/_params.py:241]: Please add help message for this parameter

**Comment by zhoxing-ms** [comment]
> [Batch] az batch pool create: Add optional --target-communication parameter
[Batch] az batch pool set: Add optional --target-communication parameter
[Batch] az batch pool show: Returned CloudPool object contains new target_node_communication_mode and current_node_communication_mode fields

Please explain the specific feature of new parameters in the history notes

File: src/azure-cli/azure/cli/command_modules/batch/_params.py (hunk 1 above)
Comment (wiboris) [reply src/azure-cli/azure/cli/command_modules/batch/_params.py]: added help

File: src/azure-cli/azure/cli/command_modules/batch/_params.py [hunk 2]
```diff
@@ -238,6 +238,12 @@ def load_arguments(self, _):
...
+        c.argument('target_node_communication_mode', options_list=['--target-communication'],
+                help="The desired node communication mode for the pool. If this element is present, it replaces the existing targetNodeCommunicationMode configured on the Pool. If omitted, any existing metadata is left unchanged.  Allowed values: classic, default, simplified.")
+
+    with self.argument_context('batch pool set') as c:
+        c.argument('target_node_communication_mode', options_list=['--target-communication'], 
+                help="The desired node communication mode for the pool. If this element is present, it replaces the existing targetNodeCommunicationMode configured on the Pool. If omitted, any existing metadata is left unchanged.  Allowed values: classic, default, simplified.")
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/batch/_params.py:246]: Could you please fix the CI issue?
```suggestion
        c.argument('target_node_communication_mode', options_list=['--target-communication'],
                           help="The desired node communication mode for the pool. If this element is present, it replaces the existing targetNodeCommunicationMode configured on the Pool. If omitted, any existing metadata is left unchanged.  Allowed values: classic, default, simplified.")
```

File: src/azure-cli/azure/cli/command_modules/batch/_params.py [hunk 3]
```diff
@@ -238,6 +238,12 @@ def load_arguments(self, _):
                 help='A space separated list of DiskEncryptionTargets. current possible values include OsDisk and TemporaryDisk.', type=disk_encryption_configuration_format)
         c.extra('image', completer=load_supported_images, arg_group="Pool: Virtual Machine Configuration",
                 help="OS image reference. This can be either 'publisher:offer:sku[:version]' format, or a fully qualified ARM image id of the form '/subscriptions/{subscriptionId}/resourceGroups/{resourceGroup}/providers/Microsoft.Compute/images/{imageName}'. If 'publisher:offer:sku[:version]' format, version is optional and if omitted latest will be used. Valid values can be retrieved via 'az batch pool supported-images list'. For example: 'MicrosoftWindowsServer:WindowsServer:2012-R2-Datacenter:latest'")
+        c.argument('target_node_communication_mode', options_list=['--target-communication'],
+                help="The desired node communication mode for the pool. If this element is present, it replaces the existing targetNodeCommunicationMode configured on the Pool. If omitted, any existing metadata is left unchanged.  Allowed values: classic, default, simplified.")
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/batch/_params.py:242]: > Allowed values: classic, default, simplified

Could we add enumerations `get_enum_type(xxx)` for these valid values

Comment (wiboris) [reply src/azure-cli/azure/cli/command_modules/batch/_params.py]: I added arg_type=get_enum_type(NodeCommunicationMode) for both

**Review by zhoxing-ms — APPROVED** [review]

**Review by zhoxing-ms — APPROVED** [review]

**Review by necusjz — APPROVED** [review]

**Pull request merged**